mlflow/models/cli.py

```diff
@@ -137,7 +138,11 @@ def build_docker(model_uri, name, install_mlflow):
 
 def _get_flavor_backend(model_uri, **kwargs):
     with TempDir() as tmp:
-        local_path = _download_artifact_from_uri(posixpath.join(model_uri, "MLmodel"),
+        if ModelsArtifactRepository.is_models_uri(model_uri):
```

Comment:
Our tests try `mlflow models predict` & `mlflow models serve` with both runs & models artifact URIs, so we should have coverage of both branches here